src/Test/Falsify/Generator/Auxiliary.hs

```diff
@@ -139,13 +149,31 @@ signedWordN = \p ->
         (if even x then Pos else Neg) $
           WordN (pred p) (x `div` 2)
 
--- | Generate fraction, shrinking towards 0
+-- | Uniform selection of fraction, shrinking towards 0
 fraction :: Precision -> Gen Fraction
 fraction p = mkFraction <$> unsignedWordN p
 
--- | Generate signed fraction, shrinking towards 0
+-- | Uniform selection of signed fraction, shrinking towards 0
 --
 -- There is no bias towards positive or negative fractions. See 'signedWordN'
 -- for more detailed discussion.
 signedFraction :: Precision -> Gen (Signed Fraction)
 signedFraction p = fmap mkFraction <$> signedWordN p
+
+{-------------------------------------------------------------------------------
+  Specialized shrinking behaviour
+-------------------------------------------------------------------------------}
+
+-- | Generator that always produces @x@ as initial value, and shrinks to @y@
+firstThen :: forall a. a -> a -> Gen a
```

Comment:
@Janiczek You might find this slightly interesting. In this PR I am changing the sample tree so that samples are marked with whether or not they have been shrunk. Most of the PR is pushing that change through, but here is an example of how this might be used; it makes it possible to define generators that always behave the same, independent from the original PRNG seed.